rs/index/src/vector/fixed_file.rs

```diff
@@ -22,33 +23,38 @@ impl<T: ToBytes + Clone> FixedFileVectorStorage<T> {
             .read(true)
             .open(file_path.clone())?;
         let mmap = unsafe { Mmap::map(&file) }?;
+        let num_vectors = transmute_u8_to_slice::<usize>(&mmap[0..8])[0];
```

Comment:
```suggestion
        let num_vectors = usize::from_le_bytes(mmap[0..8].try_into().unwrap());
```

Reply:
I think it will also be good to have a validation logic here, we we assert that length of the buffer from 8 + start to end is the same as `num_vector * num_features * size_of(T)`